src/runtime/patch/module_loader.js

```diff
@@ -0,0 +1,242 @@
+// NOTE: this is re-implementation of some core methods of
+// https://github.com/nodejs/node/blob/v10.x/lib/internal/modules/cjs/loader.js
```

Comment:
For this file, and others which re-implement Node.JS internals: Have we evaluated how much work it will take to upgrade to either `v12` or `v14`? An upgrade to one of those two versions is likely to happen in the near future for Kibana, and I want to make sure the work done here won't postpone that upgrade (we have a number of initiatives on the security team which require at least `v12`, such as `TLS 1.3` support)

Reply:
I just did a local spike to explore the static-fs with node `v12.16.1` and all the tests passed. I haven't done an extensive study but I don't anticipate big problems. We'll just need to re-sync the used files from node 10.x to 12.x and assure the fs patched apis are still up to date :)